### PR TITLE
Retain form values between login and signup page

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref #1890

### CODE REVIEW

File: resources/authgear/templates/en/web/login.html [hunk 1]
```diff
@@ -21,7 +21,7 @@
 		{{ end }}{{ end }}
 
 		<div class="twc-container-vertical pt-8 pb-5 px-6 tablet:px-8 desktop:px-8">
-			<form class="twc-container-vertical" method="post" novalidate>
+			<form id="auth-form" class="twc-container-vertical" method="post" novalidate data-controller="restore-form-local">
```
[the remark this answers is not in the document]
Comment (louischan-oursky) [reply resources/authgear/templates/en/web/login.html:25]: I changed the controller to only restore input with the same ID and name

File: authui/src/form.ts [hunk 2]
```diff
@@ -176,3 +176,92 @@ export class RestoreFormController extends Controller {
...
+      sessionStorage.removeItem(key);
+      const serializedForm: SerializedForm = JSON.parse(formString);
+
+      for (const formControl of serializedForm.controls) {
+        for (let i = 0; i < form.elements.length; ++i) {
+          const element = form.elements[i];
+          if (isSerializableFormControl(element)) {
+            if (
+              formControl.id === element.id &&
+              formControl.name === formControl.name
```
Comment (carmenlau) [inline authui/src/form.ts:257]: Should be `formControl.name === element.name`?

**Review by carmenlau — COMMENTED** [review]
Restoring the phone number seems doesn't work, no sure if it is due to the intl tel input? :(

**Comment by louischan-oursky** [comment]
I think the current approach FormRestoreLocalController is too fragile to handle intl-tel-input. Let me come up with a more robust solution. But it would require opt-in to support form restoration, so it cannot work for all forms.

**Comment by louischan-oursky** [comment]
Updated.

**Pull request merged**